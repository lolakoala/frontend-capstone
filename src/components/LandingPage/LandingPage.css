.welcome {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 75%;
  margin-left: 12.5%;
  justify-content: center;
}

.welcome p {
  font-size: 18px;
  font-weight: bold;
  color: #253031;
  text-align: center;
  margin-bottom: 20px;
}
