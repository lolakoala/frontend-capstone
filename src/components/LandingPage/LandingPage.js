import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import RaisedButton from 'material-ui/RaisedButton';
import { auth, provider } from '../../firebase.js';

class LandingPage extends Component {

  login = () => {
    auth.signInWithPopup(provider)
      .then((result) => {
        const userEmail = result.user.email;
        this.props.loginSuccess(userEmail);
      });
  }

  render() {
    const { currentUser } = this.props;
    const userKeys = Object.keys(currentUser);
    //if currentUser has one key, redirect to edit profile
    if (userKeys.length === 1) {
      return <Redirect to="/selectCity" />;
    }
    //if currentUser has > 1 key, redirect to dash
    if (userKeys.length > 1) {
      return <Redirect to="/dash" />;
    }

    return (
      <div>
        <h1>Mental Health App</h1>
        <p>Explanation of the app.</p>
        <RaisedButton label="Sign Up" onClick={this.login} />
        <RaisedButton label="Log In" onClick={this.login} />
      </div>
    );
  }
}

export default LandingPage;

LandingPage.propTypes = {
  loginSuccess: PropTypes.func,
  currentUser: PropTypes.object
};
