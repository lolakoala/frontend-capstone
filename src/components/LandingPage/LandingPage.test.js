import React from "react";
import LandingPage from './LandingPage.js';
import { shallow, configure } from "enzyme";
// import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('LandingPage', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<LandingPage
    loginSuccess={mockFn}
    currentUser={{}}
  />);
});
