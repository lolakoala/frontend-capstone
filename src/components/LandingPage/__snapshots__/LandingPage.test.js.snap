// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    currentUser={Object {}}
    loginSuccess={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Search for mental health professionals and connect with mental health buddies in your area!
</p>,
        <RaisedButton
          buttonStyle={
                    Object {
                              "height": "100px",
                              "width": "250px",
                            }
          }
          disabled={false}
          fullWidth={false}
          label="Sign Up"
          labelPosition="after"
          labelStyle={
                    Object {
                              "fontSize": "30px",
                              "marginTop": "30px",
                            }
          }
          onClick={[Function]}
          primary={false}
          secondary={true}
          style={
                    Object {
                              "marginBottom": "50px",
                            }
          }
/>,
        <RaisedButton
          buttonStyle={
                    Object {
                              "height": "100px",
                              "width": "250px",
                            }
          }
          disabled={false}
          fullWidth={false}
          label="Log In"
          labelPosition="after"
          labelStyle={
                    Object {
                              "fontSize": "30px",
                              "marginTop": "30px",
                            }
          }
          onClick={[Function]}
          primary={false}
          secondary={true}
          style={
                    Object {
                              "marginBottom": "50px",
                            }
          }
/>,
      ],
      "className": "welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for mental health professionals and connect with mental health buddies in your area!",
        },
        "ref": null,
        "rendered": "Search for mental health professionals and connect with mental health buddies in your area!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "height": "100px",
            "width": "250px",
          },
          "disabled": false,
          "fullWidth": false,
          "label": "Sign Up",
          "labelPosition": "after",
          "labelStyle": Object {
            "fontSize": "30px",
            "marginTop": "30px",
          },
          "onClick": [Function],
          "primary": false,
          "secondary": true,
          "style": Object {
            "marginBottom": "50px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "height": "100px",
            "width": "250px",
          },
          "disabled": false,
          "fullWidth": false,
          "label": "Log In",
          "labelPosition": "after",
          "labelStyle": Object {
            "fontSize": "30px",
            "marginTop": "30px",
          },
          "onClick": [Function],
          "primary": false,
          "secondary": true,
          "style": Object {
            "marginBottom": "50px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Search for mental health professionals and connect with mental health buddies in your area!
</p>,
          <RaisedButton
            buttonStyle={
                        Object {
                                    "height": "100px",
                                    "width": "250px",
                                  }
            }
            disabled={false}
            fullWidth={false}
            label="Sign Up"
            labelPosition="after"
            labelStyle={
                        Object {
                                    "fontSize": "30px",
                                    "marginTop": "30px",
                                  }
            }
            onClick={[Function]}
            primary={false}
            secondary={true}
            style={
                        Object {
                                    "marginBottom": "50px",
                                  }
            }
/>,
          <RaisedButton
            buttonStyle={
                        Object {
                                    "height": "100px",
                                    "width": "250px",
                                  }
            }
            disabled={false}
            fullWidth={false}
            label="Log In"
            labelPosition="after"
            labelStyle={
                        Object {
                                    "fontSize": "30px",
                                    "marginTop": "30px",
                                  }
            }
            onClick={[Function]}
            primary={false}
            secondary={true}
            style={
                        Object {
                                    "marginBottom": "50px",
                                  }
            }
/>,
        ],
        "className": "welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for mental health professionals and connect with mental health buddies in your area!",
          },
          "ref": null,
          "rendered": "Search for mental health professionals and connect with mental health buddies in your area!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "height": "100px",
              "width": "250px",
            },
            "disabled": false,
            "fullWidth": false,
            "label": "Sign Up",
            "labelPosition": "after",
            "labelStyle": Object {
              "fontSize": "30px",
              "marginTop": "30px",
            },
            "onClick": [Function],
            "primary": false,
            "secondary": true,
            "style": Object {
              "marginBottom": "50px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "height": "100px",
              "width": "250px",
            },
            "disabled": false,
            "fullWidth": false,
            "label": "Log In",
            "labelPosition": "after",
            "labelStyle": Object {
              "fontSize": "30px",
              "marginTop": "30px",
            },
            "onClick": [Function],
            "primary": false,
            "secondary": true,
            "style": Object {
              "marginBottom": "50px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
